File: src/game/GamePausedObserver.java
package game;

import game.hardwareinput.Command;
import game.events.Observer2;

public class GamePausedObserver implements Observer2 {
    @Override
    public void update(Object arg) {
        if (!(arg instanceof Command)) {
            return;
        }

        if(arg == Command.PAUSE){
            Bomberman.getMatch().toggleGameState();
        }
    }
}

File: src/game/engine/tasks/GamePausedObserver.kt
package game.engine.tasks

import game.Bomberman
import game.engine.tasks.observer.Observer2
import game.engine.hardwareinput.Command

class GamePausedObserver : Observer2 {
    override fun update(arg: Any?) {
        if (arg !is Command) {
            return
        }
        if (arg === Command.PAUSE) {
            Bomberman.getMatch().toggleGameState()
        }
    }
}